### DISCUSSION TITLE
Why do we need a pull up resistor on the DS18xx template?

### DISCUSSION THREAD

**Discussion by rin67630** [arendst/Tasmota#22070] [General]
I read:
"_You need to connect a 4.7K pullup resistor from data to 3.3V_"

Is that necessary for the ESP01 only?
The ESP-D1 has input-pullup in software...

**Comment by HofeBY** [comment]
Read datasheet DS18B20
https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://www.analog.com/media/en/technical-documentation/data-sheets/ds18b20.pdf&ved=2ahUKEwixzZ-p5KOIAxWTgP0HHXG8LJsQFnoECAkQAQ&usg=AOvVaw0HhbceaHCzOlfXbnNtpZZ0

**Reply by sfromis** [reply]
With a single DS18B20 on ESP8266, you can usually get away with enabling the weak internal pullup via `SetOption74`, as clearly described on the page you quoted from. That also goes for when the ESP8266 is on the non-recommendable ESP-01 board.

**Reply by rin67630** [reply]
When I understand it right, the stronger pull-up is only required if more than one sensor is on a 1 wire bus using parasitic power, right?
That does not apply when you use the VDD pin with 3,3V supply in a 3-wire bus, right?

**Reply by sfromis** [reply]
Details for when you need the stronger pullup are **already** [explained](https://tasmota.github.io/docs/DS18x20/#commands) on the page you quoted from. To quote:
> [`SetOption74`](https://tasmota.github.io/docs/Commands/#setoption74) may be used to enable/disable internal pullup when using a **single** DS18x20 sensor (for multiple sensors, you have to use an external pullup resistor). **This only works on ESP8226** ESP32 always needs external pullup

If your question about "3-wire bus" instead means connecting with one-wire data bus, along with separate vcc / gnd wires, the need for pullup does not depend on parasitic power or not; otherwise it would have been mentioned. You *always* need pullup, internal or external, as the DS18B20 can never by itself pull power level up for high, it can only pull low (open drain). Without pullup, you'd not get any signals.

What you get with parasitic power is that the DS18B20 can sip it's own power from the pulled-up data wire, meaning that it may need "more stable" pullup, and higher risk of internal pullup not being sufficient. Anyway, I'd suggest using the 3-wire setup.

**Comment by HofeBY** [comment]
@rin67630
No, you don't understand correctly that the resistance is eliminated with the two-wire connection.
It is still 4.7kOhm. See page7 & 10 of datasheet.
You can read that in the data sheet and there is nothing written about ESP there either.
What´s the problem with a 2 Cent resistor ?
